feat/IT-21-Update user profile

XiangyuZheng0802 opened the pull request:
### **Create a static page for the user profile update form, where the user can update their profile information. This page will have editable fields for the user to input the following:**

- User Name
- Language*
- Mobile
- Self Description

It will also show a disabled input for:
- email

**This task focuses on building the page layout, form fields, and basic validation of the static elements. The integration with the backend will be handled in a separate task.**

**Acceptance Criteria:**

1. **Page Layout:**

- A form should be created with the following editable fields:

    User Name (text input, optional)
    Language (dropdown/select input, required)
    Mobile (text input, optional)
    Self Description (text area, optional)

- non-editable fileds:

    email

- A submit button labeled "Save" that will be used for the integration task later.

2. **Basic Form Validation:**

    The language field should be marked as required (with a visual indicator like asterisk *).
    
    Basic frontend validation should be applied (e.g., make sure the language field is filled in).
    
    Display a generic error message below the language field if the user attempts to submit without completing it.
    
    Display a generic error message below the each field if the field fails validation.

3. **Styling:**

    Ensure the form is styled to match the general design of the application.
    
    Use chakra to keep the page consistent with the overall application design.
    
    Ensure responsiveness of this page.

4. **User Experience:**

    Include appropriate placeholders for the form fields (e.g., "Enter your name", "Select your language").
    
    Use tooltips or labels for each input field to provide a clear description.
    
    Implement simple hover and focus effects for form inputs to enhance usability.

5. **Placeholder Content:**
    
    Use placeholder text for each field where applicable.
    
    Ensure a visual design for the page that is clear and easy to navigate.

6. **No Integration:**
    
    This page is static for now; no actual data should be sent anywhere.
    
    The form submission should not trigger any backend requests. This will be handled in the integration task.

7.**New Content:**

   On the PC side, adjust the form width to one-third of the page width. 
   
   When modifying the form, use inline editing. After saving the content, display a message above the avatar indicating whether the save was successful or not.
  
![image](https://github.com/user-attachments/assets/c2af3f0a-306e-4e85-a48d-33be525a246e)
![image](https://github.com/user-attachments/assets/507aa68d-7c4a-44c1-b727-738dee91e25a)
![image](https://github.com/user-attachments/assets/8a40eae2-1f3b-408a-93c6-6850ff200a96)
![image](https://github.com/user-attachments/assets/78fa0078-138b-458e-a6fa-f46d13ed4c2b)
![image](https://github.com/user-attachments/assets/68931f1d-a27d-449e-9f62-cf3e84a892bd)
![image](https://github.com/user-attachments/assets/9413260f-5426-445d-985b-8867af53233d)
![image](https://github.com/user-attachments/assets/4b441b77-6d42-42b3-a84a-632f19329819)
<img width="1466" alt="image" src="https://github.com/user-attachments/assets/6071caa0-3b12-42fe-a914-5a265e8c95d0" />
<img width="1456" alt="image" src="https://github.com/user-attachments/assets/59446366-5ee5-4fa2-84ff-7fff23aa5ee9" />
<img width="1444" alt="image" src="https://github.com/user-attachments/assets/952cdd0a-487e-45a8-baf1-d381b6cb812f" />
<img width="1449" alt="image" src="https://github.com/user-attachments/assets/43a94a0b-d012-41c0-862b-7d249ef28053" />
<img width="1468" alt="image" src="https://github.com/user-attachments/assets/df9643bb-b1fa-428d-b7de-d637cae994fe" />
<img width="1454" alt="image" src="https://github.com/user-attachments/assets/584f2c99-0a77-435c-a9e8-84abf12dfa1d" />
<img width="1469" alt="image" src="https://github.com/user-attachments/assets/d2c3978d-a975-43a8-a927-cccbf5426cf6" />

siuLongBao reviewed (commented):
There are a few issues.
1. In the update stage, if the update is not valid, there is no indication as to which section is wrong or invalid
2. The alert is not specifying which section is wrong, there is no indication. Also, there should be no use of alert
3. The desktop version of switch language is a bit hard to use.
4. The buttons on the top has no function

XiangyuZheng0802 committed:
feat(IT-21): Update user profile

Create a static page for the user profile update form, where the user can update their profile information.
This task focuses on building the page layout, form fields, and basic validation of the static elements. The integration with the backend will be handled in a separate task.
- Implemented responsive editing: inline form editing on PC, drawer editing on mobile.
- Added individual Save buttons for each field on PC, with inline save/cancel behavior.
- Added save status feedback (success/failure message with auto-dismiss).
- Optimized form layout: adaptive width on PC (1/3 page), fixed width (343px) on mobile.
